webapp/src/hooks/useRpcProvider.ts

```diff
@@ -40,13 +43,19 @@ export function useChainlistRpcProvider(
 }
 
 export function useHarbourRpcProvider(
+	currentSettings:
+		| Pick<Partial<SettingsFormData>, "rpcUrl">
+		| undefined = undefined,
 	providerOptions: JsonRpcApiProviderOptions = DEFAULT_PROVIDER_OPTIONS,
 ): UseChainlistRpcProviderResult {
+	const settingsProvided = currentSettings !== undefined;
+	const currentRpcUrl = currentSettings?.rpcUrl;
 	const loadRpcUrl = useCallback(async (): Promise<string> => {
-		const currentSettings = await loadCurrentSettings();
-		if (currentSettings.rpcUrl) return currentSettings.rpcUrl;
-		return getRpcUrlByChainId(HARBOUR_CHAIN_ID);
-	}, []);
+		const { rpcUrl } = settingsProvided
```

Comment:
wouldn't it be simpler to check `currentRpcUrl !== undefined`?  this way you don't have to introduce `settingsProvided`. 

For me the new version quite a lot harder to understand.